charms/kubernetes_snaps.py

```diff
@@ -433,7 +468,103 @@ def configure_scheduler(extra_args_config, kubeconfig):
     with open(kube_scheduler_config_path, "w") as f:
         yaml.safe_dump(scheduler_config, f)
 
-    configure_kubernetes_service("kube-scheduler", scheduler_opts, extra_args_config)
+    config_files = {
+        scheduler_opts["config"],
+        kubeconfig,
+        kube_scheduler_config_path,
+    }
+
+    configure_kubernetes_service(
+        "kube-scheduler", scheduler_opts, extra_args_config, config_files
+    )
+
+
+@lru_cache
+def _sha256_file(config_file):
```

Comment:
let's read the contents of a file and collect its hash.  if the file is missing, it still gets a hash value